tests/utils/icu_locale_utils_tests.cpp

```diff
@@ -235,6 +235,112 @@ TEST(icu_locale_utils_test_suite, get_locale_from_str) {
     irs::string_ref expected = "de_DE_PHONEBOOK.utf-32";
     ASSERT_EQ(expected, actual);
   }
+
+  // try to break it
+  {
+    irs::string_ref locale_name = "ru_RU@let.s.break.it";
+    icu::Locale locale;
+    Unicode unicode;
+    ASSERT_TRUE(get_locale_from_str(locale_name, locale, true, &unicode));
+    ASSERT_TRUE(unicode == Unicode::NON_UTF8);
+    irs::string_ref actual = locale.getName();
+    irs::string_ref expected = "ru_RU@let.s.break.it";
+    ASSERT_EQ(expected, actual);
+  }
+
+  // try to break it
+  {
+    irs::string_ref locale_name = "ru_RU@let.s.break.it";
+    icu::Locale locale;
+    Unicode unicode;
+    ASSERT_TRUE(get_locale_from_str(locale_name, locale, false, &unicode));
+    ASSERT_TRUE(unicode == Unicode::NON_UTF8);
+    irs::string_ref actual = locale.getName();
+    irs::string_ref expected = "ru_RU";
+    ASSERT_EQ(expected, actual);
+  }
+
+  // try to break it
+  {
+    irs::string_ref locale_name = "@ru_De.utf-8";
+    icu::Locale locale;
+    Unicode unicode;
+    ASSERT_TRUE(get_locale_from_str(locale_name, locale, true, &unicode));
```

Comment:
Please add test with the same input but "new" parsing mode. That's another corner case